tests/python/gpu/test_operator_gpu.py

```diff
@@ -236,13 +236,99 @@ def test_fft():
+                                          name='batchnorm'))
+
+    # BatchNormV1 gpu
+    if 'batchnorm_v1_gpu' in batchnorm_op_list:
+      ctx_list.append({'ctx': mx.gpu(0), 'batchnorm_data': data, 'type_dict': {'batchnorm_data': np.float32}})
+      sym_list.append(mx.sym.BatchNorm_v1(fix_gamma=fix_gamma,
+                                          use_global_stats=use_global_stats,
+                                          name='batchnorm'))
+
+    # BatchNorm cpu
+    if 'batchnorm_cpu' in batchnorm_op_list:
+      ctx_list.append({'ctx': mx.cpu(0), 'batchnorm_data': data, 'type_dict': {'batchnorm_data': np.float32}})
+      sym_list.append(mx.sym.BatchNorm(fix_gamma=fix_gamma,
+                                       use_global_stats=use_global_stats,
+                                       name='batchnorm'))
+
+    # BatchNorm gpu
+    if 'batchnorm_gpu' in batchnorm_op_list:
+      ctx_list.append({'ctx': mx.gpu(0), 'batchnorm_data': data, 'type_dict': {'batchnorm_data': np.float32}})
+      sym_list.append(mx.sym.BatchNorm(fix_gamma=fix_gamma,
+                                       use_global_stats=use_global_stats,
+                                       name='batchnorm', cudnn_off=True))
+
+    # BatchNorm gpu cudnn
+    if 'batchnorm_cudnn' in batchnorm_op_list:
+      ctx_list.append({'ctx': mx.gpu(0), 'batchnorm_data': data, 'type_dict': {'batchnorm_data': np.float32}})
+      sym_list.append(mx.sym.BatchNorm(fix_gamma=fix_gamma,
+                                       use_global_stats=use_global_stats,
+                                       name='batchnorm', cudnn_off=False))
+
+    check_consistency(sym_list, ctx_list)
+
+  def test_1d_batchnorm(fix_gamma, use_global_stats):
+    data = (2, 3, 20)
+    test_batchnorm_versions_helper(batchnorm_op_list=['batchnorm_cpu',
+                                                      'batchnorm_gpu', 'batchnorm_cudnn'],
+                                   data=data,
+                                   fix_gamma=fix_gamma, use_global_stats=use_global_stats)
+
+  def test_2d_batchnorm(fix_gamma, use_global_stats):
+    data = (2, 3, 10, 10)
+    test_batchnorm_versions_helper(batchnorm_op_list=['batchnorm_v1_cpu', 'batchnorm_v1_gpu',
+                                                      'batchnorm_cpu',
+                                                      'batchnorm_gpu', 'batchnorm_cudnn'],
+                                   data=data,
+                                   fix_gamma=fix_gamma, use_global_stats=use_global_stats)
+
+  def test_3d_batchnorm(fix_gamma, use_global_stats):
+    data = (2, 3, 3, 5, 5)
+    test_batchnorm_versions_helper(batchnorm_op_list=['batchnorm_cpu',
+                                                      'batchnorm_gpu'],
+                                   data=data,
+                                   fix_gamma=fix_gamma, use_global_stats=use_global_stats)
+
+  test_1d_batchnorm(True,  False)
```

Comment:
channel_position temporarily reverted (going to put in another PR)
cudnn_off is being tested